idaes/core/surrogate/surrogate_block.py

```diff
@@ -115,9 +115,29 @@ def build_model(
                 v.setlb(lb)
                 v.setub(ub)
 
+
         # call populate block to fill-in the constraints
         surrogate_object.populate_block(self, additional_options=kwargs)
 
+        # input/output variables need to be constrained to be equal
+        # auto-created variables within surrogate object 
+        m = self.parent_block()
+        input_vars_as_enum_dict = dict(enumerate(input_vars_as_dict.keys()))
+        input_labels_enum = {v: k for k,v in input_vars_as_enum_dict.items()}
+
+        output_vars_as_dict = self.output_vars_as_dict()
+        output_vars_as_enum_dict = dict(enumerate(output_vars_as_dict.keys()))
+        output_labels_enum = {v: k for k,v in output_vars_as_enum_dict.items()}
+
+        @m.Constraint(self._input_labels)
+        def input_surrogate_ties(m, input_label):
+            return input_vars_as_dict[input_label] == self.nn.inputs[input_labels_enum.get(input_label)]
```

Comment:
This appears to be Keras/OMLT specific and will not work for other types of surrogates. Specifically, `self.nn` os not something that other types of surrogates have (e.g., ALAMO and Pysmo).

This looks like something that should go in the OMLT `populate_block` method rather than the `SurrogateBlock`.

Reply:
Modified to only apply to surrogates with a `self.nn` component.  Ultimately, this should probably move to OMLT `populate_block` as mentioned, but current version dependency does not have this functionality